components/pcloud/actions/copy-file/copy-file.js

```diff
@@ -0,0 +1,95 @@
+const common = require("../../common");
+const { pcloud } = common.props;
+const validate = require("validate.js");
+
+module.exports = {
+  key: "pcloud-copy-file",
+  name: "Copy File",
+  description:
+    "Takes one file and copies it as another file in the user's filesystem.",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    pcloud,
+    domainLocation: { propDefinition: [pcloud, "domainLocation"] },
```

Comment:
@dylburger can we move this into the app auth settings like we did for Mailgun?

Reply:
@dylburger this is good idea, to have Pipedream UI prompt for a domain or region, so that it is available in the auths variable. In my radar, besides pCloud, Zoho CRM prompts for region. her are the relevant URLs in the respective API documentation.
pCloud:  https://docs.pcloud.com/methods/oauth_2.0/authorize.html
ZohoCrm: https://www.zoho.com/crm/developer/docs/api/v2/multi-dc.html

Reply:
Yes, currently `locationid` gets returned in the access token request, which is the numeric ID corresponding to the location, so we can add this to the `$auth` variables. It looks like that might have to be mapped to `EU` / `US` here (we can't currently run those kind of custom transformations within the system that parses these variables from the OAuth flow), but you'll have access to the ID and shouldn't have to prompt the user.

I'll add that ASAP Monday.